portfolio/src/main/webapp/message-me.html

```diff
@@ -0,0 +1,52 @@
+<!DOCTYPE html>
```

Comment:
Looks like this is the same as index.html? I assume that's due to confusing messaging from me :) but, want to just merge this into the main page?

(The create-as-a-separate page plan was to make sure we were building of a base of reviewed code, but your index.html is that, so no need to separate things!)

Reply:
Hello Rob, I don't think I understand what you mean. Are you referring to the repeated code (my directory) in each one of my HTML files?

Reply:
I was referring to the repeated code -- but taking another look, it's just the header that's repeated? I think we should probably just duplicate it for now (aka don't fix it in this PR, ignore my prior comment).

----

If you wanted to push further here in a different CL, the way to avoid duplicating this code (and avoid maintaining n copies at it shows up on every page...) is generally to move to a templating framework; something that lets you generate the html programatically. (This is a common theme, and part of the reason for the comments about pulling logic out of HTML)

For example you could do:
<html>
  <div id="placeholder_header"></div>
  // ...Actual unique page content goes here....
</html>

// JS
placeholder_header.innerHTML = fetch('header_html')


This approach has some downsides thought, both in terms of flexibility, performance (this example needs an extra round trip to the server), and security. There's other ways that deal with the same issue -- IIRC the way to do it with Java8+Servlets is called "JSP"

Reply:
Thank you for the information. I will consider that and hopefully implement something to prevent writing the same HTML code for all the HTML files.